R/init.R

```diff
@@ -385,6 +396,55 @@ if (interactive() &&
         page_viewer = page_viewer
       )
 
+      #vsc-r-api
+      response_timeout <- 5
+      response_lock_file <- file.path(dir_extension, "response.lock")
+      response_file <- file.path(dir_extension, "response.log")
```

Comment:
Putting the response file here might be problematic if user has multiple vscode instances, all sharing the same response file.

Reply:
oh I thought we changed that to the session dir.... :thinking:

Reply:
Oh dear: https://github.com/Ikuyadeu/vscode-R/pull/408/files#diff-f3fe9c4fe673e665c8df5914c3b9c9a0R65

half right.

Reply:
The file should be put to the reader, R session dir in this case, whereas the request file is read by vscode-R extension.

Reply:
...how was that even working... at all?

Reply:
Oh wait I see. Watcher dir, not session dir?

Reply:
So I think I have thought about this a while ago. It's kind of okay, since all responses are synchronous and addins are interactive dev tools?

Pretty much impossible for a human to launch two in a way that will create a problem?

Reply:
Oh wait but I have seen an addin that polled the active document context in `{sinew}`. Boo.

Reply:
So maybe this need some rejigging on the request side of things too. If `request.wd` was the session dir, we could know to write back to a response file in that session dir?

Reply:
ha but of course that can't work for the 'attach' handler because we don't know our session dir.

So maybe we leave that as is, but all calls to `request_response` from the R side include the session dir? which the response uses to write back to?

Reply:
Ok I think I have successfully moved the response files to the session dir. That was good actually things are a bit simpler now, and I think it shook out a bug with the file setup.